### ISSUE TITLE
WC_Order_Item_Meta deprecated, but suggested replacement lacks its abilities

### ISSUE THREAD

**Issue by DavidAnderson684** [woocommerce/woocommerce#14623]
### EXPLANATION OF THE ISSUE

In one of my extensions, I call WC_Order_Item_Meta::display() in order to get a fragment to include in a print-out of the order summary. This is now apparently deprecated (since 3.0.0) - https://docs.woocommerce.com/wc-apidocs/class-WC_Order_Item_Meta.html - and we are directed to wc_display_item_meta().

However, wc_display_item_meta() does not operate over the same list of items. It calls WC_Order_Item_Product::get_formatted_meta_data(), which uses wc_is_attribute_in_product_name() to filter out some of the items. As a result, in my particular use case, I get an empty result, where I used to get back something printable that included the chosen variation.

The wording of the deprecation notice suggests that it's being deprecated because nothing in core uses it. So perhaps the intended solution for extension authors is that we clone the deprecated class before it vanishes from core entirely in a later release?

In summary: though the class is deprecated, I'm unclear on what I'm meant to do about that. The replacement is different, so is this just unfortunate and we have to carry more code? Or something else?

**Comment by mikejolley** [comment]
This is actually filterable now if you want to disable the variation meta handling core uses https://github.com/woocommerce/woocommerce/blob/8bc715ca59d8b3d612b3d41b1febca3ed1fea213/includes/wc-attribute-functions.php#L335

But yes, nothing in core uses it and it will eventually be removed. `wc_display_item_meta()` contains the logic we feel is best going forward.
